model/testing/tests/test_verification_benchmarking.py

```diff
@@ -0,0 +1,61 @@
+# ICON4Py - ICON inspired code in Python and GT4Py
+#
+# Copyright (c) 2022-2024, ETH Zurich and MeteoSwiss
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+import functools
+from typing import Optional
+
+import numpy as np
+import pytest
+from numpy.typing import NDArray
+
+from icon4py.model.testing import helpers
+
+
+BASE_DTYPE = np.int64
+
+
+def incr_func(
+    field: NDArray[BASE_DTYPE],
+    increment: int,
+):
+    field += increment
+
+
+def verify_field(
+    field: NDArray[BASE_DTYPE],
+    increment: int,
+    base_value: int,
+):
+    np.testing.assert_allclose(field, base_value + increment)
+
+
+def test_verification_benchmarking_infrastructure(benchmark: Optional[pytest.FixtureRequest]):
+    base_value = 1
+    field = np.array((base_value * np.ones((), dtype=BASE_DTYPE)))
+
+    increment = 6
+
+    helpers.run_verify_and_benchmark(
+        functools.partial(incr_func, field=field, increment=increment),
+        functools.partial(verify_field, field=field, increment=increment, base_value=base_value),
+        benchmark_fixture=None,  # no need to benchmark this test
+    )
```

Comment:
As I said in a previous review, instead of passing the real `benchmark` fixture you should pass a mock (please read the documentation at https://devdocs.io/python~3.13/library/unittest.mock#unittest.mock.Mock) here with the `enabled` attribute set to True/False and then assert if it has been called or not. Something like:

```suggestion
    benchmark = mock.Mock(enabled=False)
    
    helpers.run_verify_and_benchmark(
        functools.partial(incr_func, field=field, increment=increment),
        functools.partial(verify_field, field=field, increment=increment, base_value=base_value),
        benchmark_fixture=benchmark,
    )
    assert not benchmark.called

    benchmark.enabled = True
    helpers.run_verify_and_benchmark(
        functools.partial(incr_func, field=field, increment=increment),
        functools.partial(verify_field, field=field, increment=increment, base_value=base_value),
        benchmark_fixture=benchmark,
    )
    assert benchmark.called
```
This is only an example to show how to use it, don't take it as a real suggestion. Try to fit something like that in the test in a way that makes sense and tests all relevant code paths.